### PR TITLE
Refactored modal_aero_amicphys_intr up-to-date with refactor-maint-2.0

### PR DESCRIPTION
**Pull request by huiwanpnnl**
This PR replaces PR #49:

- Revisions suggested by PR reviewers were implemented.
- Refactored code was rebased onto `refactor-maint-2.0` in order to be able to use subroutine `min_max_bound` in `chemistry/utils/mam_support.F90`
- Commit history is clean.

**Testing on Compy:**

Commands used to run all tests:

```
cd /qfs/people/wanh895/codes/e3sm_mam4_refactor_amicphys_subareas/
bash mam4_refactor_scripts/compare_baselines_all_tests.sh  \
     -b huiwanpnnl/refactor-amicphys-intr2 \
     -t ALL_TESTS_refactor-amicphys-intr2
```

Commands used for checking the results:

```
cd /compyfs/wanh895/e3sm_scratch//ALL_TESTS_refactor-amicphys-intr2
./cs.status.comp_ALL_TESTS_refactor-amicphys-intr2_mam4_org_v2_baselines -s
```

The results are

```
  PASS ERP_P64x1.ne4pg2_oQU480.F2010.compy_intel
  PASS ERS.ne4pg2_oQU480.F2010.compy_intel
  PASS PEM_P64x1.ne4pg2_oQU480.F2010.compy_intel
  PASS PET.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS_D_Ln5_P32x1.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS_D.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS_Ly1.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS.ne30pg2_EC30to60E2r2.F2010.compy_intel
```

### CODE REVIEW

File: components/eam/src/chemistry/modal_aero/modal_aero_amicphys.F90 [hunk 1]
```diff
@@ -90,615 +108,243 @@ subroutine modal_aero_amicphys_intr(                             &
...
 
    logical :: do_cond, do_rename, do_newnuc, do_coag
-   logical :: iscldy_subarea(maxsubarea)
-
-   character(len=fieldname_len+3) :: fieldname
-   character(len=6)   :: tmpch6a, tmpch6c
-   character(len=200) :: tmp_str
-
-   real (r8) :: pdel_fac
+   logical :: lcopy(gas_pcnst)
```
Comment (singhbalwinder) [inline components/eam/src/chemistry/modal_aero/modal_aero_amicphys.F90]: `lcopy` seems not being used here.

File: components/eam/src/chemistry/modal_aero/modal_aero_amicphys.F90 [hunk 2]
```diff
@@ -90,615 +108,243 @@ subroutine modal_aero_amicphys_intr(                             &
...
 
+   ! Column integrals need to be initialized with zeros
+      q_coltendaa = 0.0_r8 
+   qqcw_coltendaa = 0.0_r8
 
+   call amicphys_diags_init( do_cond, do_rename, do_newnuc, do_coag )
 
-      adjust_factor_pbl_ratenucl = newnuc_adjust_factor_pbl
+   !====================================================
+   ! Get saturation mixing ratio
```
Comment (singhbalwinder) [inline components/eam/src/chemistry/modal_aero/modal_aero_amicphys.F90:251]: May be mention it explicitly that `qv_sat` is the output here. `ev_sat` is also an output but it is not used anywhere.

**Review by singhbalwinder — APPROVED** [review]
Awesome job, Hui! This looks great! I have some very minor comments but it looks great overall!

File: components/eam/src/chemistry/modal_aero/modal_aero_amicphys.F90 (hunk 1 above)
Comment (huiwanpnnl) [reply components/eam/src/chemistry/modal_aero/modal_aero_amicphys.F90]: Thanks for the very careful review. `lcopy` is removed in [fcdc573](https://github.com/eagles-project/e3sm_mam4_refactor/pull/53/commits/fcdc573c9dbf8bd518c32f218a0f617d4b954ab9).

File: components/eam/src/chemistry/modal_aero/modal_aero_amicphys.F90 (hunk 2 above)
Comment (huiwanpnnl) [reply components/eam/src/chemistry/modal_aero/modal_aero_amicphys.F90:251]: Comments on the `intent` of the arguments have been added in [fcdc573](https://github.com/eagles-project/e3sm_mam4_refactor/pull/53/commits/fcdc573c9dbf8bd518c32f218a0f617d4b954ab9).

**Comment by huiwanpnnl** [comment]
Test again on Compy after commit [fcdc573](https://github.com/eagles-project/e3sm_mam4_refactor/pull/53/commits/fcdc573c9dbf8bd518c32f218a0f617d4b954ab9):

Commands used to run all tests:
```
cd /qfs/people/wanh895/codes/e3sm_mam4_refactor_amicphys_subareas/
bash mam4_refactor_scripts/compare_baselines_all_tests.sh \
     -b huiwanpnnl/refactor-amicphys-intr2 \
     -t ALL_TESTS_refactor-amicphys-intr2_20230105
```
Commands used for checking the results:
```
cd /compyfs/wanh895/e3sm_scratch//ALL_TESTS_refactor-amicphys-intr2_20230105
./cs.status.comp_ALL_TESTS_refactor-amicphys-intr2_20230105_mam4_org_v2_baselines -s
```
The results are
```
  PASS ERP_P64x1.ne4pg2_oQU480.F2010.compy_intel
  PASS ERS.ne4pg2_oQU480.F2010.compy_intel
  PASS PEM_P64x1.ne4pg2_oQU480.F2010.compy_intel
  PASS PET.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS_D_Ln5_P32x1.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS_D.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS_Ly1.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS.ne30pg2_EC30to60E2r2.F2010.compy_intel
```

**Comment by huiwanpnnl** [comment]
@singhbalwinder , if you are happy with the update, could you help merge the PR? (After the merge I will submit the next PR for another refactored subroutine in `modal_aero_amicphys.F90`). Thanks.

**Comment by singhbalwinder** [comment]
Thanks, Hui for addressing my comments and merging this PR.

**Pull request merged**